clients/src/main/java/org/apache/kafka/common/requests/ListOffsetRequest.java

```diff
@@ -88,7 +87,7 @@ public Builder setTargetTimes(Map<TopicPartition, Long> partitionTimestamps) {
         public ListOffsetRequest build() {
             short version = version();
             if (version < minVersion) {
-                throw new ObsoleteBrokerException("The broker is too old to send this request.");
+                throw new UnsupportedVersionException("The broker is too old to understand this request.");
```

Comment:
Good point.  I added information about the API being called and the version sent (versus required) to the exception.

For the exception below, it should not actually be reachable unless there is a programming error, so it is better off as a RuntimeException rather than an UnsupportedVersionException.